virtual_ecosystem/models/soil/carbon.py

```diff
@@ -12,36 +12,52 @@
 from virtual_ecosystem.core.constants import CoreConsts
 from virtual_ecosystem.models.soil.constants import SoilConsts
 from virtual_ecosystem.models.soil.env_factors import (
+    EnvironmentalEffectFactors,
     calculate_environmental_effect_factors,
     calculate_leaching_rate,
     calculate_temperature_effect_on_microbes,
 )
 
 
 @dataclass
-class MicrobialBiomassLoss:
-    """A data class to store the various biomass losses from microbial biomass."""
+class MicrobialChanges:
+    """Changes due to microbial uptake, biomass production and losses."""
 
-    maintenance_synthesis: NDArray[np.float32]
-    """Rate at which biomass must be synthesised to balance losses [kg C m^-3 day^-1].
-    """
-    pom_enzyme_production: NDArray[np.float32]
-    """Rate at which POM degrading enzymes are produced [kg C m^-3 day^-1]."""
-    maom_enzyme_production: NDArray[np.float32]
-    """Rate at which MAOM degrading enzymes are produced [kg C m^-3 day^-1]."""
-    necromass_decay_to_lmwc: NDArray[np.float32]
-    """Rate at which biomass is lost to the LMWC pool [kg C m^-3 day^-1]."""
-    necromass_decay_to_pom: NDArray[np.float32]
-    """Rate at which biomass is lost to the POM pool [kg C m^-3 day^-1]."""
-
-
-# TODO - This function should probably be shortened, leaving as is for the moment as a
-# sensible split will probably be more obvious once more is added to this function.
+    lmwc_uptake: NDArray[np.float32]
```

Comment:
are you always talking about rates of change? At the moment, most of these names can be easily interpreted as stocks, for example the `microbe`